deploy/storage-provisioner/Dockerfile

```diff
@@ -12,6 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM scratch
-COPY out/storage-provisioner storage-provisioner
-CMD ["/storage-provisioner"]
+FROM k8s.gcr.io/debian-base-amd64:v2.0.0
```

Comment:
Can this stay `scratch`?

Reply:
I think we need to provide a build container for the copied file, right now it cheerfully builds it on the host ?

Use the normal MINIKUBE_BUILD_IN_DOCKER shenaningans, I don't see what it has do with path though...

Reply:
> Can this stay `scratch`?

Can I resolve this conversation ? as it's clear now in the slack channel conversation that we cannot use scratch ?

Reply:
As long as you build static binaries (using CGO_ENABLED=0), then `scratch` should be fine.

What issues are you seeing ? We know that a _dynamic_ binary doesn't work with scratch...
But that was probably not the intention, but just a side-effect when redoing the make earlier.

I added the build container in #6257 if you want to have a look.